### PR TITLE
♻️ Statically format filter pages.

### PR DESCRIPTION
**Pull request by matthiasrohmer**
Filters to `pages-transformed` currently to be able to retest.

### CODE REVIEW

File: platform/lib/build/pageTransformer.js [hunk 1]
```diff
@@ -107,23 +110,111 @@ class PageMinifier {
           let html = canonicalPage.contents.toString();
           html = scope.minifyPage(html, canonicalPage.path);
 
-          const ampPath = canonicalPage.relative.replace('.html', '.amp.html');
-          const optimizedHtml = await scope.optimize(html, ampPath);
```
Comment (matthiasrohmer) [inline platform/lib/build/pageTransformer.js]: Needs to be reactivated.

File: platform/lib/build/pageTransformer.js [hunk 2]
```diff
@@ -107,23 +110,111 @@ class PageMinifier {
...
 
+          for (const page of [canonicalPage, ampPage, ...filteredPages]) {
+            this.push(page);
+          }
+
+          scope._log.success(`Transformed ${canonicalPage.path}`);
           callback();
         }))
-        .pipe(gulp.dest('./'));
+        .pipe(gulp.dest('./pages-transformed'));
```
Comment (matthiasrohmer) [inline platform/lib/build/pageTransformer.js]: Needs to be `./` again.

File: platform/lib/routers/pages.js [hunk 3]
```diff
@@ -231,27 +216,26 @@ if (!config.isDevMode()) {
...
-        // Check if there's a manually filtered variant ...
-        const manualRequestPath = requestPath.replace('.html', `.${format}.html`);
+        // Check if there's a manually filtered variant and respect AMP variant
+        if (requestPath.endsWith('.amp.html')) {
+          const manualRequestPath = requestPath.replace('.amp.html', `.${format}.amp.html`);
+        } else {
+          const manualRequestPath = requestPath.replace('.html', `.${format}.html`);
+        }
+
         if (await fs.existsSync(utils.project.pagePath(manualRequestPath))) {
```
Comment (sebastianbenz) [inline platform/lib/routers/pages.js]: use `existsAsync` (defined in the same file)

File: platform/lib/common/filteredPage.js [hunk 4]
```diff
@@ -43,10 +43,11 @@ const FILTERED_ROUTES = [
  */
 function filterPage(format, dom, force) {
   if (!isAvailable(format, dom) && !force) {
-    return;
+    return false;
```
Comment (sebastianbenz) [inline platform/lib/common/filteredPage.js]: return either dom as well for consistency or return true at the end

File: platform/lib/build/pageTransformer.js [hunk 5]
```diff
@@ -141,14 +141,16 @@ class PageTransformer {
...
-    // Skip pages that have been manually filtered and therfore have a path like
+    // Skip pages that have been manually filtered and therefore have a path like
     // - guides-and-tutorials/index.websites.html
     // - guides-and-tutorials/index.email.amp.html
-    if (path.match(/\.(websites|stories|ads|email)(\.amp)?\.html/)) {
-      return true;
+    // But still the matching format should be filtered, therefore pass it back
+    const format = path.match(/\.(websites|stories|ads|email)(\.amp)?\.html/);
+    if (format) {
+      return format[1];
```
Comment (sebastianbenz) [inline platform/lib/build/pageTransformer.js]: Is it intentional that only this path returns a format?

Comment (sebastianbenz) [reply platform/lib/build/pageTransformer.js]: vs true/false

File: platform/lib/build/pageTransformer.js [hunk 6]
```diff
@@ -116,15 +119,108 @@ class PageMinifier {
...
+    return filteredPages;
+  }
+
+  /**
+   * @param  {String} html
+   * @return {String}
+   */
+  filterHtml(html, format, force) {
+    const dom = cheerio.load(html);
+    if (filterPage(format, dom, force)) {
```
Comment (sebastianbenz) [inline platform/lib/build/pageTransformer.js]: nit: using  an early exit avoids indentation

Comment (sebastianbenz) [reply platform/lib/build/pageTransformer.js]: it's also weird that only one code path returns a value

File: platform/lib/build/pageTransformer.js (hunk 5 above)
Comment (matthiasrohmer) [reply platform/lib/build/pageTransformer.js]: Yes and no, those should have been two separate methods as now.

File: platform/lib/build/pageTransformer.js (hunk 6 above)
Comment (matthiasrohmer) [reply platform/lib/build/pageTransformer.js]: `filterPage` now returns `true|false`.

File: platform/lib/build/pageTransformer.js [hunk 7]
```diff
@@ -116,14 +120,123 @@ class PageMinifier {
...
+        continue;
+      }
+
+      for (const format of FORMATS) {
+        if (this._hasManualFiltered(page, format)) {
+          continue;
+        }
+
+        const filteredHtml = this.filterHtml(html, format);
+        if (filteredHtml) {
```
Comment (sebastianbenz) [inline platform/lib/build/pageTransformer.js]: this is always true as you're now returning the input html string.

Comment (matthiasrohmer) [reply platform/lib/build/pageTransformer.js]: Thanks, returns `undefined` for unfiltered HTML again now, which is already gracefully handled by both `pageTransformer.js` and `pages.js` router.

**Pull request merged**